pkg/rtc/types/interfaces.go

```diff
@@ -86,6 +86,8 @@ type Participant interface {
 	AddSubscribedTrack(st SubscribedTrack)
 	RemoveSubscribedTrack(st SubscribedTrack)
 	SubscriberPC() *webrtc.PeerConnection
+	AddSubscribedTrackToStreamAllocator(st SubscribedTrack)
+	RemoveSubscribedTrackFromStreamAllocator(st SubscribedTrack)
```

Comment:
I would prefer it if we didn't have to add new participant level APIs that exposes these concepts. How about this instead?

SubscribedTrack exposes a `OnStart` or `OnBind` callback, when when a downtrack is bound, it forwards it to the subscribed track.

Then within `AddSubscribedTrack`, we could listen to that and then start it to the underlying transport. 

For removal, it seems that we would handle it in `RemoveSubscribedTrack`? they are called at the same time.

Reply:
@davidzhao Something like this? - https://github.com/livekit/livekit-server/pull/261/commits/f2b0d30b143af37f6ab00d3eb61d6eb37edc869e

Reply:
looks great!